### PR TITLE
 Add CSV export feature

### PR DESCRIPTION
**Pull request by a-mccarthy**
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

fixes: https://github.com/goharbor/harbor/issues/17081

### CODE REVIEW

**Comment by a-mccarthy** [comment]
@AllForNothing If you have a minute, can you please review?

One question I have is the only way to download through the Event log window? It looks like once you click the download option, the CVS is removed.

**Comment by AllForNothing** [comment]
@a-mccarthy 
_is it the only way to download through the Event log window?_
It's the only way on UI. And you can also call the download API directly
_It looks like once you click the download option, the CVS is removed._
This is by design. (It helps save space in Harbor and improve performance)

**Comment by AllForNothing** [comment]
@a-mccarthy After the performance test, we found that exporting (all) multiple projects would cause performance issues.
So we decided to disable exporting (all) multiple projects.  And related PRs are Here:https://github.com/goharbor/harbor/pull/17265(for back-end),https://github.com/goharbor/harbor/pull/17267(for UI)
Once these PRs merged, I will build a new Harbor for you to update this DOC PR

**Comment by AllForNothing** [comment]
@a-mccarthy 
This env(https://10.202.250.189) has been upgraded to the latest version. And you can start to update this PR now.

Updates:
1. now, you can only export(select) one project at one time, and you can only click the `export button` once in 30 seconds
2. Add a new  type of permission for the export function, only `system admin, project admin, maintainer, developer` have this exporting permission:
image

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 1]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
 
 
 When an OCI image index is successfully scanned, the summary of the vulnerability report for the OCI image index is aggregated from the individual scan results of the the artifacts referenced by the index. The vulnerability report will show both sets of statistics.
+
+## Export Vulnerability Scans
+
+As a project administrator, you are able to export vulnerability scan results of one or more of your projects as CVS and download it from your Harbor instance.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:75]: As a `project admin or maintainer or developer`
only able to export one project at one time

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 2]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
...
+
+1. Log in to the Harbor interface with an account that has at least project administrator privileges.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:77]: at lease `developer`

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 3]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
...
+1. Go to **Projects** and select a project.
+1. Select the checkbox for the projects you would like to export.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:79]: should remove this step

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 4]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
...
+1. In the Export CVE window you can apply any filters you want to the data included in the CVS.
+
+  * **Repositories:** Filters by repository name. Enter multiple repository names in a comma separated list, repositories match a name `repo*`,or ** for all repositories in the projects.
+  * **Tags:** Filters by tags. Enter multiple tags in a comma separated list, tags match a name `tag*`,or ** for all tags in the projects.
+  * **Labels:** Filters by labels. Select multiple labels from the drop down.
+  * **CVE IDs:** Filters by CVE number. Enter comma separated list of CVE IDs.
+
+    ![Export CVE filter options](../../../img/filter-export-vulnerabilities.png)
+
+1. Click **Export**. Harbor will begin creating the CVS file with your exported CVE details. The generated CVE file will be named `csv_file_` and the current date and time formatted as YYYYMMDDHHMM. For example, `csv_file_202207281130.csv` was created on 07/28/2022 at 11:30.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:93]: formatted as YYYYMMDDHHMMSS.  It is now accurate to the second

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 5]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
...
+
+1. Download the CVE file from the Event Log window. Expand the Event Log window by clicking on **Event Log** from the Harbor interface.
+1. Click the download icon next to the CSV file you want to download.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:96]: Once finishing download, the related execution record will be removed

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md [hunk 6]
```diff
@@ -69,3 +69,30 @@ When scanning an OCI image index, Harbor will send scan requests for each of the
...
+As a project administrator, you are able to export vulnerability scan results of one or more of your projects as CVS and download it from your Harbor instance.
+
+1. Log in to the Harbor interface with an account that has at least project administrator privileges.
+1. Go to **Projects** and select a project.
+1. Select the checkbox for the projects you would like to export.
+1. Select **Action** and then **Export CVE(s)**.
+
+  ![Export CVE](../../../img/export-vulnerabilities.png)
+
+1. In the Export CVE window you can apply any filters you want to the data included in the CVS.
```
Comment (AllForNothing) [inline docs/administration/vulnerability-scanning/scan-individual-artifact.md:84]: typo:  in the CSV, not CVS

**Comment by AllForNothing** [comment]
@a-mccarthy You can refer to this https://confluence.eng.vmware.com/display/HARBOR/Test+cases+for+CVE+data+export
to help you better understand the CVE export function

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md (hunk 6 above)
Comment (a-mccarthy) [reply docs/administration/vulnerability-scanning/scan-individual-artifact.md:83]: updated!

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md (hunk 1 above)
Comment (a-mccarthy) [reply docs/administration/vulnerability-scanning/scan-individual-artifact.md]: updated

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md (hunk 3 above)
Comment (a-mccarthy) [reply docs/administration/vulnerability-scanning/scan-individual-artifact.md]: removed

File: docs/administration/vulnerability-scanning/scan-individual-artifact.md (hunk 4 above)
Comment (a-mccarthy) [reply docs/administration/vulnerability-scanning/scan-individual-artifact.md]: updated

**Comment by a-mccarthy** [comment]
@AllForNothing thanks for the updates! i think i've updated the docs to align with the new functionality. can you review again?

**Review by AllForNothing — APPROVED** [review]
LGTM

**Review by prahaladdarkin — APPROVED** [review]
looks good to me

**Review by chlins — APPROVED** [review]
lgtm

**Pull request merged**